core/DataTable/Filter/Sort.php

```diff
@@ -243,6 +231,51 @@ public function filter($table)
             }
         }
 
-        $table->sort(array($this, $methodToUse), $this->columnToSort);
+        $this->sort($table, $methodToUse);
     }
+
+    /**
+     * Sorts the DataTable rows using the supplied callback function.
+     *
+     * @param string $functionCallback A comparison callback compatible with {@link usort}.
+     * @param string $columnSortedBy The column name `$functionCallback` sorts by. This is stored
+     *                               so we can determine how the DataTable was sorted in the future.
+     */
+    private function sort(DataTable $table, $functionCallback)
+    {
+        $table->setTableSortedBy($this->columnToSort);
+
+        $rows = $table->getRowsWithoutSummaryRow();
+
+        // get column value and label only once for performance tweak
+        $values = array();
+        foreach ($rows as $key => $row) {
+            $values[$key] = array($this->getColumnValue($row), $row->getColumn('label'));
```

Comment:
On 25k rows `getColumnValue` was executed > 1 mio times. Now we do it only 25k times which brings big performance boost